Add AudioStore - audio equivalent to ImageStore

mariuszskon opened the pull request:
Unblocks #281 and #282

Progress tracked at #322 

## Description
Currently, CC Catalog only handles images. In particular, the backend is still solely focused on images, blocking work on ingesting data of other types, such as audio. This PR adds some of the glue code in the form of AudioStore which is the audio equivalent to ImageStore. Hopefully this lays down some foundations that can later be implemented in the database etc.

## Other information
I inherited AudioStore directly from ImageStore, but it is probably a better idea to have some base class instead.

## Checklist
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `master` branch of the repository.
- [x] My commit messages follow [best practices][best_practices].
- [x] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

<!-- Make sure you read and understand the following attestation. -->
## Developer Certificate of Origin
```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

mariuszskon committed:
Add AudioStore

mariuszskon committed:
Add tests for AudioStore (naive copy of ImageStore tests)

mariuszskon committed:
Fix AudioStore tests to ensure columns correctly checked

mathemancer commented:
Assigning myself to keep this on my radar

mathemancer commented:
@mariuszskon Please keep in mind that the Audio Provider API Script integrations are blocked until we get the infrastructure work done (we'll need to set up some tables in a DB, for example)

mathemancer reviewed (commented):
This is a great start!  For context, here's a list of the pieces of info we'd like to store for audio providers (if available):

- Title (could be name of recording)
- Creator (a.k.a. artist, musician, etc.)
- License
- Date
- Collection/Album
- File format (e.g. mp3)
- Type (e.g. music, podcast, lecture)
- Duration
- File size
- Genre
- Language
- Samplerate
- Bitdepth
- Channels
- Tags

I'd love it if you would create an issue in this repo to help us keep track of this work, and keep notes about what the differences between this `AudioStore` class and the `ImageStore` class should be.

mariuszskon commented:
@mathemancer #322

mariuszskon committed:
Add more fields to AudioStore

Add album, file format, duration, genre, language, samplerate, bitdepth, channels.

mariuszskon committed:
Add audio_type field to AudioStore

mathemancer commented:
Closing this until we're ready to work on audio.  Thank you for the effort on this, @mariuszskon !
